nixos/modules/services/misc/matrix-synapse.nix

```diff
@@ -176,548 +169,497 @@ in {
+                      "http"
+                      "manhole"
+                      "metrics"
+                      "replication"
+                    ];
+                    default = "http";
+                    example = "metrics";
+                    description = ''
+                      The type of the listener, usually http.
+                    '';
+                  };
+
+                  tls = mkOption {
+                    type = types.bool;
+                    default = true;
+                    example = false;
+                    description = ''
+                      Whether to enable TLS on the listener socket.
+                    '';
+                  };
+
+                  x_forwarded = mkOption {
                     type = types.bool;
+                    default = false;
+                    example = true;
                     description = ''
-                      Should synapse compress HTTP responses to clients that support it?
-                      This should be disabled if running synapse behind a load balancer
-                      that can do automatic compression.
+                      Use the X-Forwarded-For (XFF) header as the client IP and not the
+                      actual client IP.
+                    '';
+                  };
+
+                  resources = mkOption {
+                    type = types.listOf (types.submodule {
+                      options = {
+                        names = mkOption {
+                          type = types.listOf (types.enum [
+                            "client"
+                            "consent"
+                            "federation"
+                            "keys"
+                            "media"
+                            "metrics"
+                            "openid"
+                            "replication"
+                            "static"
+                          ]);
+                          description = ''
+                            List of resources to host on this listener.
+                          '';
+                          example = [
+                            "client"
+                          ];
+                        };
+                        compress = mkOption {
+                          type = types.bool;
+                          description = ''
+                            Should synapse compress HTTP responses to clients that support it?
+                            This should be disabled if running synapse behind a load balancer
+                            that can do automatic compression.
+                          '';
+                        };
+                      };
+                    });
+                    description = ''
+                      List of HTTP resources to serve on this listener.
                     '';
                   };
                 };
               });
+              default = [ {
+                port = 8448;
```

Comment:
This is a port synapse has historically used for federation. It's a little confusing to default to it for client as well. Without extra steps, the server must be reachable over <servername>:8448 with tls. and it's a waste to use more ports, but it's something to think about.

Reply:
I updated the port from 8008 because I saw `tls` defaulting to true. I'd be happy to update the default to `8008` with `tls` disabled, which should be the more prominent scenario with regards to reverse proxying.. Does that sound good to you?

Reply:
Migrated to a default setup for reverse proxying via localhost.